src/sage/schemes/elliptic_curves/ell_field.py

```diff
@@ -1634,7 +1657,26 @@ def isogenies_prime_degree(self, l=None, max_l=31):
                 L = [ZZ(d) for d in l]
 
         from .isogeny_small_degree import isogenies_prime_degree
-        return sum([isogenies_prime_degree(self, d) for d in L], [])
+        if cyclic_after is None:
+            return sum([isogenies_prime_degree(self, d) for d in L], [])
+        else:
+            dom = cyclic_after.domain()
+            if cyclic_after.codomain() != self:
+                raise ValueError("The codomain of `cyclic_after` must be the same curve")
+            if not cyclic_after.is_separable():
+                raise ValueError("The `cyclic_after` isogeny must be separable")
+            ret = []
+            for d in L:
+                for phi in isogenies_prime_degree(self, d):
```

Comment:
This would call the current method recursively, I don't see a reason for this. Calling the function `isogenies_prime_degree` directly just mimicks the code from L1661.

Reply:
Oh wait, I see, sorry for that. I understand the logic now